### ISSUE TITLE
`Failed to authenticate with Quick Unlock: Polkit authorization failed` error on Ubuntu 24.04

### ISSUE THREAD

**Issue by EmilEmilchen** [keepassxreboot/keepassxc#11316]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
In the latest snapshot (2.8.0 c1a66a8) version of KeePassXC I get an error when trying to use QuickUnlock (Polkit) on Kubuntu 24.04.1:
`Failed to authenticate with Quick Unlock: Polkit authorization failed`

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Install Kubuntu 24.04.1 LTS
2. Download v2.8.0 c1a66a8 (or - as far I can tell - any v2.8.0 pre release) AppImage or flatpak
3. Launch KeePassXC
4. Open or Create a database
5. Unlock it and try unlocking it using QuickUnlock

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
The polkit (`PolicyKit1 KDE Agent`) is supped to appear and allow for easy access to the database.

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
I get an error `Failed to authenticate with Quick Unlock: Polkit authorization failed` in the GUI:

![The error](https://github.com/user-attachments/assets/b37994e1-efc7-45aa-973a-7b9c698af54d)

## Context
[NOTE]: # ( Give us any additional information you may have. )
I'm - as stated below - running KUbuntu 24.04.1 LTS under Wayland with an NVIDIA GPU (just mentioning this because it's been quite a pain to get everything working nicely with it and Wayland).
I'm 99% sure though this isn't the problem here. Even when switching to my CPU's integrated graphics or X11 or both I still have the same problem.
I was unable to find any additional debug information, cli doesn't show anything at all and as far as I can tell the debug/logging functionality of KeePassXC is rather limited (I could be mistaken, I definitely couldn't figure out how to get any more info).
The `PolicyKit1 KDE Agent` dialog does show up and work just fine anywhere else. 
The most similar issue I was able to find is #9981 but that's definitely not the problem I'm encountering as I can confidently say that the dialog wasn't raised at all, not just not brought to the front.
I even tried this on a clean install of Kubuntu 24.04.1 LTS where I just installed `plasma-workspace-wayland` and nvidia drivers as well as configured Wayland to use the Nvidia GPU.

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.8.0
Revision: c1a66a8

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux
Desktop Env: KDE
Windowing System: Wayland (error definitely also occurs with X11)

**Comment by droidmonkey** [comment]
https://github.com/keepassxreboot/keepassxc/issues/5991#issuecomment-2171750016

But we need to provide better notice when this file is not present. The quick unlock page also incorrectly shows even though polkit process was never initiated.

**Comment by thatguyatgithub** [comment]
I'm using Debian with fprintd `1.94.4` and Kde with Wayland. I've checked that the user had enrolled successfully and works as for dbus goes. 
It seems I'm experiencing the same issue as the reporter, in the sense that the `Failed to authenticate with Quick Unlock: Polkit authorization failed` is triggered immediately with no errors shown in any logs at polkit (in debug), systemd's, X's or keepass'.

I might be missing something on the setup, I understand that polkit is not requiring any extra communication. 

Does anyone successfully managed to get it to work on a configuration similar to this Linux setup?

**Comment by EmilEmilchen** [comment]
I had it working on debian in the meantime as well, on a testing system. Just to be sure, you did run
```
sudo wget -P /usr/share/polkit-1/actions https://raw.githubusercontent.com/keepassxreboot/keepassxc/refs/heads/develop/share/linux/org.keepassxc.KeePassXC.policy.in && sudo chmod 644 /usr/share/polkit-1/org.keepassxc.KeePassXC.policy.in
```
Just asking because you didn't specifically mention it.

**Comment by thatguyatgithub** [comment]
> I had it working on debian in the meantime as well, on a testing system. Just to be sure, you did run
> 
> ```
> sudo wget -P /usr/share/polkit-1/actions https://raw.githubusercontent.com/keepassxreboot/keepassxc/refs/heads/develop/share/linux/org.keepassxc.KeePassXC.policy.in && sudo chmod 644 /usr/share/polkit-1/org.keepassxc.KeePassXC.policy.in
> ```
> 
> Just asking because you didn't specifically mention it.

Hi Emil! 
I've changed the chmod command to include the `/actions` subdir path from your quote and still didn't seem to change anything. Setting polkit in debug kind of shows no hint of any polkit ruling specific for KeePassXC, not sure if it should still, shoudl it?

```
Dec 14 11:26:01 x1 systemd[1]: Starting polkit.service - Authorization Manager...
Dec 14 11:26:01 x1 polkitd[148828]: Started polkitd version 125
Dec 14 11:26:01 x1 polkitd[148828]: Loading rules from directory /etc/polkit-1/rules.d
Dec 14 11:26:01 x1 polkitd[148828]: Loading rules from directory /usr/share/polkit-1/rules.d
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/50-default.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/60-libvirt.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.Flatpak.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.GeoClue2.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.NetworkManager.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.bolt.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.fwupd.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.freedesktop.packagekit.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/org.rnd2.cpupower-gui.rules
Dec 14 11:26:01 x1 polkitd[148828]: Loaded and executed script in file /usr/share/polkit-1/rules.d/systemd-networkd.rules
Dec 14 11:26:01 x1 polkitd[148828]: Finished loading, compiling and executing 10 rules
Dec 14 11:26:01 x1 systemd[1]: Started polkit.service - Authorization Manager.
Dec 14 11:26:01 x1 polkitd[148828]: Acquired the name org.freedesktop.PolicyKit1 on the system bus
```

The `make` does not make any reference nor appears to carry any PolKit file, rule, policy or anything of the like, maybe that's the part missing?

**Comment by EmilEmilchen** [comment]
Sorry for the typo. If the file is there with correct permissions I don't really have any idea what could be going on either but have you tried just using the prebuild from [here](https://snapshot.keepassxc.org/)? Don't really know why that would help but it's what I had done when it used to work on debian machine.

**Comment by balor** [comment]
@EmilEmilchen @thatguyatgithub 
Shouldn't the policy have `.policy` extension to be loaded on action?

**Comment by EmilEmilchen** [comment]
@balor 
Yes it absolutely should just have the `.policy` extension without the `.in`. Good chance that was the problem I think. Maybe give it a try with the other filename @thatguyatgithub

**Comment by dothyt** [comment]
came across this post while having some trouble setting up my polkit for keepassxc on fedora 42 with KDE. Eventually manage to make it work. In my case, it was SELinux blocking polkit reading the `.policy` file I created. had to change the selinux security context with `chcon` to be the same as other `.policy` file

some of the debugging process I went through for people having different setup:
1. run `pkaction | grep keepass` to see if the `action_id` has been registered in polkit
2. run `man polkit` to see what are the directory `.policy` can be placed
3. remove `--no-debug` in `/usr/lib/systemd/system/polkit.service` `ExecStart` section, check `journalctl` for logs
